x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -261,7 +263,48 @@ export const CredentialsLogic = kea<
         flashAPIErrors(e);
       }
     },
-    // TODO onApiTokenChange from ent-search
+    onApiTokenChange: async () => {
+      const { myRole } = AppLogic.values;
+      const {
+        id,
+        name,
+        engines,
+        type,
+        read,
+        write,
+        access_all_engines: accessAllEngines,
+      } = values.activeApiToken;
+
+      const data: IApiToken = {
```

Comment:
I don't think your tests above cover this logic enough. 

i.e., what happens if a token is sent with ApiTokenTypes.Admin and activeApiToken AND it has data.read and data.write values ... those should NOT be sent on the server request. I think we should have a test that asserts that condition. That's not covered, currently.

Reply:
Yeah, I don't know. If it's part of the behavior of a unit I think it should be covered in a test, personally. That let's someone:

A) Easily understand what this code is doing why
B) Refactor it if needed and know they haven't changed the behavior

As is, it's not clear from the spec what is happening or why. And since the condition isn't covered by the spec, I could easily inadvertently break this units interface during a refactor.

Reply:
Totally fair 👍 new tests are in https://github.com/elastic/kibana/pull/81519/commits/b34d7300de4e66791bcdd73f9d1cdacb65c041d5 - let me know if just testing search/admin is good enough for you (w/ a note about private), or if you want to include a full test for private keys as well.

Reply:
This looks good to me